src/debugSession/BrightScriptDebugSession.ts

```diff
@@ -1686,6 +1693,203 @@ export class BrightScriptDebugSession extends BaseDebugSession {
         return results;
     }
 
+    private failedToLoadCompletionPaths = new Set<string>();
+
+    protected async completionsRequest(response: DebugProtocol.CompletionsResponse, args: DebugProtocol.CompletionsArguments, request?: DebugProtocol.Request) {
+        this.logger.log('completionsRequest', args, request);
+        // this.sendEvent(new LogOutputEvent(`completionsRequest: ${args.text}`));
+        // this.sendEvent(new OutputEvent(`completionsRequest: ${args.text}\n`, 'stderr'));
+
+        try {
+            let provideGlobalCallables = false;
+            if ((args.column !== args.text.length + 1)) {
+                response.body = {
+                    targets: []
+                };
+                return this.sendResponse(response);
+            }
+            let variablePath: string[] = [];
+            if (!args.text.trim()) {
+                variablePath = [''];
+            } else if (args.text.endsWith('.')) {
+                variablePath = util.getVariablePath(args.text.slice(0, -1));
+            } else {
+                variablePath = util.getVariablePath(args.text);
+            }
+
+            let completions: DebugProtocol.CompletionItem[] = [];
+            if (variablePath) {
+                let parentVariablePath: string[];
+                // If the last character is a period, then pull completions for the parent variable before the period
+                if (args.text.endsWith('.')) {
+                    parentVariablePath = variablePath;
+                } else {
+                    parentVariablePath = variablePath.slice(0, variablePath.length - 1);
+                }
+
+                if (parentVariablePath.length === 0) {
+                    parentVariablePath = [''];
+                }
+
+                if (parentVariablePath.length === 1 && parentVariablePath[0] === '') {
+                    provideGlobalCallables = true;
+                }
+
+                let parentVariable = this.findVariableByPath(Object.values(this.variables), parentVariablePath, args.frameId);
+
+                if ((!parentVariable || parentVariable.childVariables.length === 0) && !this.failedToLoadCompletionPaths.has(parentVariablePath.join('.'))) {
```

Comment:
include frame id in this failed key. 

Also do we even need this extra layer? Seems like the number of requests we end up making isn't really much different unless we key off of where the path becomes invalid. 

Lastly if they do anything other then a variable look up this will get cleared with the variables cache to. Might just not be worth the squeeze.